Sources/NIO/PendingWritesManager.swift

```diff
@@ -478,6 +479,9 @@ extension PendingWritesManager {
                 }
 
                 oneResult = try triggerOneWriteOperation(self.currentBestWriteMechanism)
+                // We may have become unwritable re-entrantly.
+                becameUnwritable = (becameUnwritable || !self.isWritable)
```

Comment:
So the downside of this is that we are now repeatedly polling the atomic `isWritable` variable. This is a _technically correct_ solution, so we may need to merge it if we can't find a better one, but I'd like to look at alternative options.

One thing I'm wondering about is whether we can defer writability changes that occur while we're already in `triggerWriteOperations`. Essentially, if we're in `triggerWriteOperations`, perhaps `add` should _never_ issue a writability notification.

Another option that preserves these writability notifications (which, to be clear, I believe are useful) is to have a state variable updated by `add` if it does emit a `false` writability state update. That state variable can be checked by this code at the bottom of the loop. This is probably the easiest to implement and simplest to understand: maybe we should do that?

What do you think @glbrntt?

Reply:
> So the downside of this is that we are now repeatedly polling the atomic `isWritable` variable. 

Agreed, it's far from ideal!

> One thing I'm wondering about is whether we can defer writability changes that occur while we're already in `triggerWriteOperations`. Essentially, if we're in `triggerWriteOperations`, perhaps `add` should _never_ issue a writability notification.
> 
> Another option that preserves these writability notifications (which, to be clear, I believe are useful) is to have a state variable updated by `add` if it does emit a `false` writability state update. That state variable can be checked by this code at the bottom of the loop. This is probably the easiest to implement and simplest to understand: maybe we should do that?

I think it's better to preserve the writability notifications, I'll give this second approach a shot.